Sources/App/Core/Search.swift

```diff
@@ -124,7 +124,34 @@ enum Search {
             .orderBy(score, SQLDirection.descending)
             .orderBy(packageName, SQLDirection.ascending)
             .offset(offset)
-            .limit(pageSize + 1)  // fetch one more so we can determine `hasMoreResults`
+            .limit(limit)
+            .select
+    }
+
+    static func query(_ database: Database,
+                      _ terms: [String],
+                      page: Int,
+                      pageSize: Int) -> SQLSelectBuilder? {
+        guard let db = database as? SQLDatabase else {
+            fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
+        }
+
+        // page is one-based, clamp it to 0-based offset
+        let offset = ((page - 1) * pageSize).clamped(to: 0...)
+        let limit = pageSize + 1  // fetch one more so we can determine `hasMoreResults`
+
+        guard let inner = packageMatchQuery(on: database,
+                                            terms: terms,
+                                            offset: offset,
+                                            limit: limit) else {
+            return nil
+        }
+
+        return db.select()
+            .column("*")
+            .from(
+                SQLAlias(SQLGroupExpression(inner), as: SQLIdentifier("t"))
+            )
```

Comment:
This looks a bit silly now with just an empty select wrapper but I wasn't sure how nesting works with SQLKit and we need this as a wrapper around the `UNION` selects.